regressiontests/regression_test.go

```diff
@@ -0,0 +1,391 @@
+//go:build regression
+// +build regression
+
+package regressiontests
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/bitcoin-sv/block-headers-service/config"
+	"github.com/bitcoin-sv/block-headers-service/domains"
+	"github.com/bitcoin-sv/block-headers-service/internal/chaincfg"
+	"github.com/bitcoin-sv/block-headers-service/transports/http/endpoints/api/headers"
+	"github.com/bitcoin-sv/block-headers-service/transports/http/endpoints/api/merkleroots"
+	"github.com/bitcoin-sv/block-headers-service/transports/http/endpoints/api/tips"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const localHTTPServerURL = "http://localhost:8080"
+const whatsonchainAPIURL = "https://api.whatsonchain.com/v1/bsv/main/chain/tips"
+
+type WhatsOnChainForkTip struct {
+	Height    int    `json:"height"`
+	Hash      string `json:"hash"`
+	BranchLen int    `json:"branchlen"`
+	Status    string `json:"status"`
+}
+
+var dbEngine string
+
+func init() {
+	flag.StringVar(&dbEngine, "dbEngine", "sqlite", "The database engine to use in tests (postgres or sqlite)")
+}
+
+func TestApplicationIntegration(t *testing.T) {
```

Comment:
This is minor, because I don't have any suggestions,
but I wonder if there is a possibility to somehow write it in such a way, that the setup part won't take 80% of the code in a test, because it makes this test harder to understand what actually is a goal of it.

Reply:
I tried to simplify and make it more readable - please reverify